src/utils/HTTPTransport.ts

```diff
@@ -0,0 +1,68 @@
+enum METHODS {
+  GET = 'GET',
+  POST = 'POST',
+  PUT = 'PUT',
+  PATCH = 'PATCH',
+  DELETE = 'DELETE',
+}
+
+type RequestData = Record<string, string | number>;
+
+type RequestOptions = {
+  method?: METHODS;
+  headers?: Record<string, string>;
+  timeout?: number;
+  data?: unknown;
+};
+
+function queryStringify(data: RequestData) {
+  if (!data) {
+    return '';
+  }
+  return Object.entries(data).reduce(
+    (acc, [key, value], index, arr) =>
+      `${acc}${key}=${value}${index < arr.length - 1 ? '&' : ''}`,
+    '?'
+  );

```

Comment:
Этот код выглядит довольно сложно, я предложу более простую реализацию
```suggestion
  return '?' + Object
    .entries(data)
    .map(([key, value]) => `${key}=${value}`)
    .join("&")

```